lib/flutter_vlc_player.dart

```diff
@@ -1,15 +1,337 @@
+    /// parentHeight are the width and height of the parent perhaps as defined by a LayoutBuilder.
+    @required this.aspectRatio,
+    /// This is the initial URL for the content. This also must be provided but [VlcPlayerController] implements
+    /// [VlcPlayerController.setStreamUrl] method so this can be changed at any time.
+    @required this.url,
+    /// Before the platform view has initialized, this placeholder will be rendered instead of the video player.
+    /// This can simply be a [CircularProgressIndicator] (see the example.)
+    this.placeholder,
+  });
+
+  @override
+  _VlcPlayerState createState() => _VlcPlayerState();
+}
+
+class _VlcPlayerState extends State<VlcPlayer> with AutomaticKeepAliveClientMixin {
+  VlcPlayerController _controller;
+  int videoRenderId;
+  bool playerInitialized = false;
+
+  @override
+  bool get wantKeepAlive => true;
+
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return AspectRatio(
+      aspectRatio: widget.aspectRatio,
+      child: Stack(
+        children: <Widget>[
+          Offstage(offstage: playerInitialized, child: widget.placeholder ?? Container()),
+          Offstage(
+            offstage: !playerInitialized,
+            child: _createPlatformView(),
+          ),
+        ],
+      ),
+    );
+  }
+
+  Widget _createPlatformView() {
+    if (Platform.isIOS) {
+      return UiKitView(
+        viewType: "flutter_video_plugin/getVideoView",
+        onPlatformViewCreated: _onPlatformViewCreated
+      );
+    } else if (Platform.isAndroid) {
+      return AndroidView(
+        viewType: "flutter_video_plugin/getVideoView",
+        hitTestBehavior: PlatformViewHitTestBehavior.transparent,
+        onPlatformViewCreated: _onPlatformViewCreated
+      );
+    }
+
+    throw new Exception("flutter_vlc_plugin has not been implemented on your platform.");
+  }
+
+  void _onPlatformViewCreated(int id) async {
+    _controller = widget.controller;
+    _controller.registerChannels(id);
+
+    _controller.addListener((){
+      if(!mounted) return;
+
+      // Whenever the initialization state of the player changes,
+      // it needs to be updated. As soon as the Flutter part of this library
+      // is aware of it, the controller will fire an event, so we're okay
+      // to update this here.
+      if(playerInitialized != _controller.initialized) setState(() {
+        playerInitialized = _controller.initialized;
+      });
+    });
+
+    // Once the controller has clients registered, we're good to register
+    // with LibVLC on the platform side.
+    if (_controller.hasClients) {
+      await _controller._initialize(widget.url);
+    }
   }
+
+  @override
+  void deactivate() {
+    _controller.dispose();
+    playerInitialized = false;
+    super.deactivate();
+  }
+
+  @override
+  void dispose() {
+    _controller.dispose();
+    super.dispose();
+  }
+}
+
+class VlcPlayerController {
```

Comment:
@NBTX is there any particular reason to place all classes in one file? I think it violates SRP - if we want to simplify imports we can create library, but that will make migration a bit more complex. What do you think?

Reply:
If you merge this PR, @solid-software  you will need to version the library to 2.0.0 as the code is breaking change and backwards compatibility is broken. I tested this by importing this library in my own code and I had to refactor the VLC widget.

Reply:
@mitchross I changed the version code.
I think I will follow @solid-software's advice and use multiple files. When I refactored initially one file made sense, but obviously it no longer does.

Reply:
Cool. let me know when you are ready for a retest!

Reply:
i wanna test things and give thumbs up to things

Reply:
@NBTX When are you going to update?